src/main/java/gov/scot/payments/paymentfile/ilf/ILFFileParser.java

```
@@ -0,0 +1,117 @@
+    }
+
+    private void validateCSVRecord(CSVRecord record) throws FileParseException {
+
+        if(record.size() > MAX_ROW_SIZE || record.size() < MIN_ROW_SIZE){
+            throw new FileParseException("Invalid row count for record "+ record.toString());
+        }
+
+        int[] requiredIndex = {SORT_CODE_INDEX, ACCOUNT_NUM_INDEX, AMOUNT_INDEX};
+        for(int index : requiredIndex){
+            if(record.get(index) == null){
+                throw new FileParseException("Could not find value for index "+ index);
+            }
+        }
+    }
+
+
+    private boolean emptyCSVRecord(CSVRecord record) {
+
+        var valueItr = record.iterator();
+        while(valueItr.hasNext()){
+            String value = valueItr.next();
+            if(value != null){
+                return false;
+            }
+        }
+        return true;
+    }
+
+
+    private boolean validateBACSRecord(ILFBACSPayment payment) throws FileParseException {
+
+        if(!Utils.isValidSortCode(payment.getSortCode())){
+            throw new FileParseException("Invalid sort code.");
+        }
+        if(!Utils.isValidAccountNumber(payment.getAccountNumber())){
+            throw new FileParseException("Invalid account number.");
+        }
+        if(!Utils.isValidMoneyAmount(payment.getAmount())){
+            throw new FileParseException("Invalid amount");
+        }
+        return true;
+    }
+
+    public List<ILFBACSPayment> generatePaymentList(InputStream paymentFileStream) throws IOException, FileParseException {
+
+        var reader = new InputStreamReader(paymentFileStream);
+        List<ILFBACSPayment> parsed = new ArrayList<>();
+
+        CSVFormat format = CSVFormat.RFC4180
+                .withTrim()
+                .withNullString("")
+                .withIgnoreEmptyLines(true);
+
+        try (var csvParser = new CSVParser(reader, format)) {
+
+            for (CSVRecord csvRecord : csvParser) {
+
+                if (emptyCSVRecord(csvRecord)) {
```

not sure this is needed, just let the record creation fail if its empty rather than skipping empty rows

---

In the example files they sent over, some of them contain rows with all nulls (,,,,,,) at the end of the file. Trying to capture this an skip rather than fail the whole file.